File: BaekJoon/Queue/2164.py
from collections import deque
import sys

N = int(sys.stdin.readline())

li = [x for x in range(1,N+1)]
queue = deque(li)

while len(queue) != 1:
    queue.popleft()
    if len(queue) == 1:
        break
    queue.append(queue.popleft())
    
print(queue.pop())

File: BaekJoonSwift/BaekJoonSwift/Queue & Deque/2164.swift
import Foundation

let n = Int(readLine()!)!
var queue = Array(1...n)
var front = 0
var rear = n

while rear - front > 1 {
    front += 1
    if rear - front == 1 {
        break
    }
    queue.append(queue[front])
    front += 1
    rear += 1
}

print(queue[front])
